pkg/oc/cli/admin/release/new.go

```diff
@@ -538,6 +549,32 @@ func (o *NewOptions) Run() error {
 		is.Annotations = make(map[string]string)
 	}
 
+	if o.CoreOSURL != "" {
+		coreosBuild, err := GetCoreOSBuild(context.TODO(), o.CoreOSURL, o.CoreOSVersion)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(o.Out, "Using CoreOS build %s\n", coreosBuild.BuildID)
+		digestedImage := fmt.Sprintf("%s@%s", coreosBuild.OSContainer.Image, coreosBuild.OSContainer.Digest)
+		if digestedImage == "@" {
+			return fmt.Errorf("No oscontainer in CoreOS build")
+		}
+
+		// Hardcoded, this name was chosen in one of the machine-config-operator PRs
+		// and added to the release payload by Clayton.
+		o.Mappings = append(o.Mappings, Mapping{Source: "machine-os-content",
+			Destination: digestedImage})
+
+		// And inject the full build metadata - primarily useful for
+		// "bootimages" i.e. AMIs/qcow2/etc.
+		serializedBuild, err := json.Marshal(coreosBuild)
+		if err != nil {
+			return err
+		}
+		// This is written as a label in the final image
+		is.Annotations[coreOSBootImageLabel] = string(serializedBuild)
```

Comment:
Ahh OK, I see now why you wanted to also deserialize the qcow2s. Though... isn't there a way to have this data be a proper configmap instead of JSON-in-a-label? We were discussing this in https://github.com/openshift/machine-config-operator/issues/183 right? (And including the pkglist, etc...).

Anyway, totally fine doing it this way for now too!